pkg/cloudprovider/providers/azure/azure_util_vmss.go

```diff
@@ -160,79 +160,88 @@ func (ss *scaleSet) updateCache() error {
 }
 
 // getCachedVirtualMachine gets virtualMachine by nodeName from cache.
+// If the node is not belonging to any scale sets, it returns cloudprovider.InstanceNotFound.
 func (ss *scaleSet) getCachedVirtualMachine(nodeName string) (scaleSetVMInfo, error) {
 	ss.cacheMutex.Lock()
 	defer ss.cacheMutex.Unlock()
 
-	getVMFromCache := func(nodeName string) (scaleSetVMInfo, error) {
+	getVMFromCache := func(nodeName string) (scaleSetVMInfo, bool) {
+		glog.V(8).Infof("Getting scaleSetVMInfo for %q from cache %v", nodeName, ss.cache)
 		for scaleSetName := range ss.cache {
 			for _, vm := range ss.cache[scaleSetName] {
 				if vm.NodeName == nodeName {
-					return vm, nil
+					return vm, true
 				}
 			}
 		}
 
-		return scaleSetVMInfo{}, cloudprovider.InstanceNotFound
+		return scaleSetVMInfo{}, false
 	}
 
-	vm, err := getVMFromCache(nodeName)
-	if err == nil {
+	vm, found := getVMFromCache(nodeName)
+	if found {
 		return vm, nil
 	}
 
 	// Known node not managed by scale sets.
 	if ss.availabilitySetNodesCache.Has(nodeName) {
+		glog.Infof("Found node %q in availabilitySetNodesCache", nodeName)
```

Comment:
If it always hit cache, I would suggest set as `glog.V(10).Infof` to reduce info logs

Reply:
Good catch. Thanks